src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java

```diff
@@ -106,6 +109,16 @@ public String getIamMfaSerialNumber() {
         return iamMfaSerialNumber;
     }
 
+    @NonNull
+    public Integer getStsTokenDuration() {
+        return stsTokenDuration == null ? DescriptorImpl.defaultStsTokenDuration : stsTokenDuration;
+    }
+
+    @DataBoundSetter
+    public void setStsTokenDuration(@NonNull Integer stsTokenDuration) {
+        this.stsTokenDuration = stsTokenDuration.equals(DescriptorImpl.defaultStsTokenDuration) ? null : stsTokenDuration;
```

Comment:
With this logic you can relax the `@NonNull` condition, which is only used by static analysis and cannot be enforced by Jelly and write:

```
this.stsTokenDuration = stsTokenDuration == null || stsTokenDuration.equals(DescriptorImpl.defaultStsTokenDuration) ? null : stsTokenDuration;
```

Reply:
Thanks for the feedback. I've added the improvements for how the method handles null values now